lib/full-site-editing/template-parts.php

```diff
@@ -60,6 +60,35 @@ function gutenberg_register_template_part_post_type() {
 }
 add_action( 'init', 'gutenberg_register_template_part_post_type' );
 
+/**
+ * Registers the 'template_part_type' taxonomy.
+ */
+function gutenberg_register_template_part_section_taxonomy() {
+	if ( ! gutenberg_is_fse_theme() ) {
+		return;
+	}
+
+	register_taxonomy(
+		'section',
+		array( 'wp_template_part' ),
+		array(
+			'public'            => false,
+			'hierarchical'      => false,
+			'labels'            => array(
+				'name'          => __( 'Sections', 'gutenberg' ),
+				'singular_name' => __( 'Section', 'gutenberg' ),
+			),
+			'query_var'         => false,
+			'rewrite'           => false,
+			'show_ui'           => false,
+			'_builtin'          => true,
+			'show_in_nav_menus' => false,
+			'show_in_rest'      => false,
```

Comment:
Im not 100% sure on which settings we really want currently.  For now this is the same for how the `wp_theme` term is set up.  Maybe we would want this `public` and `show_ui` in the future? 🤔

Reply:
I think these are good callouts. I'm not certain either.

Unless someone has strong use cases, I'd defer to hiding the ui, especially since there's little to no documentation about the subject. If I'm not mistaken, it's also something we could easily introduce in the future without causing breaking changes.

Reply:
>  I'd defer to hiding the ui, especially since there's little to no documentation about the subject. If I'm not mistaken, it's also something we could easily introduce in the future without causing breaking changes.

Thats what I was thinking as well.  Start with these settings (as long as they make sense to everyone) and update them in the future if necessary.

Reply:
One argument in favor of `show_ui` is that it might make development and debugging easier, since we can more easily inspect the values or manually add new ones there etc. Other than that I think this should be hidden. Also I was expecting that we'd need `show_in_rest` to be `true` down the road?

Reply:
We want this to be essentially hidden terms we select through the editor UI alone (similar to how post formats are a taxonomy) so we should disable most things.